pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -504,6 +504,9 @@ type Template struct {
 
 	// Metrics are a list of metrics emitted from this template
 	Metrics *Metrics `json:"metrics,omitempty" protobuf:"bytes,35,opt,name=metrics"`
+
+	// Memoize allows templates to use outputs generated from already executed templates
+	Memoize *Memoize `json:"memoize,omitempty"`
```

Comment:
How do we handle the co-existence of this and `argo resubmit --memoized`?

Reply:
I don't think there's any overlap, since the resubmit flag is a CLI feature and the memoization option here is set in a workflow spec. The resubmit flag also only uses memoization for the same instance of a workflow, while this feature allows different workflows to share the cache.